extendeddatacrossword.go

```diff
@@ -324,3 +309,11 @@ func noMissingData(input [][]byte) bool {
 	}
 	return true
 }
+
+func (eds *ExtendedDataSquare) computeSharesRoot(shares [][]byte, i uint) []byte {
+	tree := eds.createTreeFn()
+	for cell, d := range shares {
+		tree.Push(d, SquareIndex{Cell: uint(cell), Axis: i})
+	}
+	return tree.Root()
+}
```

Comment:
👍🏼 This is a bit orthogonal remark, but I always wondered if the API of the underlying tree should not take a list of leaves directly and return the root instead of pushing these in a loop instead. I think everywhere we are using the tree, we always have all leaves at hand at once anyways.

Reply:
Yeah probably! Intuitively, I don't think this is a huge overhead though (worst-case amortized linear cost, so order-optimal), so maybe lower-priority.